### PR TITLE
Issue 8 main page

### PR DESCRIPTION
**Pull request by MumuCarrot**
### Hero section
The main banner must present the platform’s mission and niche (e.g., Об’єднуємо крафтових виробників та інноваторів)
Visually group relevant product categories: ВИРОБНИЦТВО, УПАКОВКА, etc.
Include a strong CTA button: Долучитися до платформи

image
image
image

### Нові учасники (New Participants)
Section must display cards with: logo/image, company name, and label (e.g., Нові компанії)
Button Усі підприємства → must navigate to the full directory

image
image
image
image
image
image
image

### Value Proposition Section
Section titled Майданчик для тих, хто втілює свої ідеї в життя
CTA button Долучитися below the statement

image
image
image

### Target Audience (Для кого)
Show grid of cards/icons for each audience type:
- Виробники
- Стартапи
- Громадські організації
- Інші
All icons/titles must be accessible and evenly spaced

image
image
image

### Benefits Section (Чому варто)
image
image
image

### Footer
image
image
image

### CODE REVIEW

File: frontend/src/components/ParticipantCard/participantCard.jsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ *   alt="Company name"
+ *   title="My Company"
+ *   location="New York"
+ *   uid="12345"
+ *   recentlyUpdated={true}
+ * />
+ */
+function ParticipantCard({bcgImgSrc, ppImgSrc, alt, title, location, uid, className, recentlyUpdated}) {
+    const { user } = useAuth();
+    const [isFollowed, setFollow] = useState(false);
```
Comment (mehalyna) [inline frontend/src/components/ParticipantCard/participantCard.jsx:42]: Consider initializing `isFollowed` from a prop or context if the follow state can be persisted or pre-fetched, instead of always starting as false.

File: frontend/src/components/ParticipantCard/participantCard.jsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    const [isRecentlyUpdated] = useState(!!recentlyUpdated);
```
Comment (mehalyna) [inline frontend/src/components/ParticipantCard/participantCard.jsx:43]: There’s no need to store `isRecentlyUpdated` in state if it’s not going to change during the component’s lifecycle. Use a plain variable instead:
```
const isRecentlyUpdated = !!recentlyUpdated;
```

File: frontend/src/components/ParticipantCard/participantCard.jsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    const navigate = useNavigate();
+    const [companyLink] = useState(`/profile/company/${uid}`);
```
Comment (mehalyna) [inline frontend/src/components/ParticipantCard/participantCard.jsx:45]: Storing `companyLink` in state is unnecessary since it’s derived from a prop and doesn’t change. Use a plain variable instead.

File: frontend/src/components/ParticipantCard/participantCard.jsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    return (
+        <div className={`participant-card ${className || ''}`}>
```
Comment (mehalyna) [inline frontend/src/components/ParticipantCard/participantCard.jsx:48]: If `className` is undefined or empty, this can produce an extra space.
Recommend using a library like `clsx` for cleaner conditional class management, or:
```
<div className={`participant-card${className ? ` ${className}` : ''}`}>
```

File: frontend/src/components/ParticipantCard/participantCard.jsx
Comment (mehalyna) [inline frontend/src/components/ParticipantCard/participantCard.jsx:1]: **Line 49, 57, 66, 72, 82**

Setting tabIndex={-1} on links prevents keyboard users from accessing these links, which hurts accessibility.
Unless there is a strong UX reason, it’s better to omit tabIndex={-1} for links.

File: frontend/src/components/ParticipantCard/participantCard.jsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                        <span>Інші послуги</span>
+                    </Link>
+                </div>
+                <div className={'participant-card--text-container'}>
+                    <h3 className={'participant-card--title'}>
+                        <Link to={companyLink}>
+                            <span>{title}</span>
+                        </Link>
+                    </h3>
+                    <p className={'participant-card--location'}>{location}</p>
```
Comment (mehalyna) [inline frontend/src/components/ParticipantCard/participantCard.jsx:76]: If location data can be missing or null, add a fallback value or conditional rendering.

File: frontend/src/components/ParticipantCard/participantCard.jsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                    <div>
+                        <Button
+                            className={'participant-card--services-btn'}
+                            onClick={() => navigate(companyLink)}
+                        >
+                            Послуги
+                        </Button>
+                    </div>
+                    {user && (
+                        <FollowStar value={isFollowed} setter={setFollow} />
```
Comment (mehalyna) [inline frontend/src/components/ParticipantCard/participantCard.jsx:88]: Consider passing a unique identifier as a prop to `FollowStar` (e.g., `participantId={uid}`) so it can manage its own state or trigger API calls for following/unfollowing.

File: frontend/src/pages/Home/home.jsx [hunk 7]
```diff
@@ -1,13 +1,256 @@
...
+ * @example
+ * <HomeSection color="blue" className="custom-class">
+ *   <h2>Welcome</h2>
+ *   <p>This is the home page.</p>
+ * </HomeSection>
+ */
+function HomeSection({ color, className, children }) {
+    return (
+        <section className={`home--section__${color}`}>
+            <div className={`container home--container ${className}`}>
```
Comment (mehalyna) [inline frontend/src/pages/Home/home.jsx:28]: If `className` is undefined, this will result in an unwanted "undefined" in the class list. Use:
```
<div className={`container home--container${className ? ' ' + className : ''}`}>
```

File: frontend/src/pages/Home/home.jsx
Comment (mehalyna) [inline frontend/src/pages/Home/home.jsx:1]: **Line 53, 74, 98**

Using array index as a key can cause rendering issues if items are reordered or removed. Prefer a unique identifier (e.g., `item.uid` for participants).

File: frontend/src/pages/Home/home.jsx [hunk 8]
```diff
@@ -1,13 +1,256 @@
...
+        {title: 'Інновації та тренди', description: 'Будьте в курсі останніх новинок та технологій галузі'},
+        {title: 'Співпраця та синергія', description: 'Об\'єднуйтесь, щоб творити нове та ділитися досвідом'},
+        {title: 'Розвиток та масштабування', description: 'Знаходьте нових партнерів, клієнтів та ринки збуту'},
+        {title: 'Підтримка та знання', description: 'Отримуйте консультації, експертну допомогу та доступ до освітніх ресурсів'}
+    ];
+
+    const navigate = useNavigate();
+
+    // State for changing the link text in the “New participants section”
+    const [isMobile, setIsMobile] = useState(window.innerWidth <= 769);
```
Comment (mehalyna) [inline frontend/src/pages/Home/home.jsx:155]: Direct access to `window` may cause issues during server-side rendering. Consider checking for window existence or using a custom hook for responsive logic.

File: frontend/src/pages/Home/home.jsx [hunk 9]
```diff
@@ -1,13 +1,256 @@
...
+                    <h2 className={'home--title participants--title__flex'}>
+                        Нові учасники
+                    </h2>
+                    <Link to={'/companies'} className={'link-right-arrow'}>
+                        <p className={'participants--link link__underline'} aria-label="Переглянути всі підприємства">
+                            {isMobile ? 'Всі' : 'Всі підприємства'}
+                        </p>
+                    </Link>
+                </div>
+                <NewParticipantGrid data={newParticipantsData} />
```
Comment (mehalyna) [inline frontend/src/pages/Home/home.jsx:215]: If participant data can be empty, handle empty state in the grid component (e.g., show a message or fallback UI).

**Review by mehalyna — CHANGES_REQUESTED** [review]
Add test coverage for all grid components, especially for empty and edge cases.

File: frontend/src/pages/Home/home.jsx
Comment (MumuCarrot) [reply frontend/src/pages/Home/home.jsx:1]: Done✅

File: frontend/src/pages/Home/home.jsx (hunk 8 above)
Comment (MumuCarrot) [reply frontend/src/pages/Home/home.jsx:155]: Created a resizing hook with useIsMobile to solve this problem✅

File: frontend/src/pages/Home/home.jsx (hunk 9 above)
Comment (MumuCarrot) [reply frontend/src/pages/Home/home.jsx:215]: Created a GenericGrid component to solve this problem✅

File: frontend/src/components/ParticipantCard/participantCard.jsx (hunk 6 above)
Comment (MumuCarrot) [reply frontend/src/components/ParticipantCard/participantCard.jsx:88]: Implemented✅

File: frontend/src/components/ParticipantCard/participantCard.jsx (hunk 5 above)
Comment (MumuCarrot) [reply frontend/src/components/ParticipantCard/participantCard.jsx:76]: Implemented✅

File: frontend/src/components/ParticipantCard/participantCard.jsx (hunk 2 above)
Comment (MumuCarrot) [reply frontend/src/components/ParticipantCard/participantCard.jsx:43]: Reasonable, changed✅

File: frontend/src/components/ParticipantCard/participantCard.jsx (hunk 1 above)
Comment (MumuCarrot) [reply frontend/src/components/ParticipantCard/participantCard.jsx:42]: Implemented✅

**Comment by MumuCarrot** [comment]
Should I do something about the tests or will they fix themselves as soon as Maxim Zymyn makes a Merge pull request? Because I didn't pull the main for this branch.

File: frontend/src/components/FollowStar/followStar.jsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+                </svg>
```
[the remark this answers is not in the document]
Comment (MumuCarrot) [reply frontend/src/components/FollowStar/followStar.jsx:33]: I do. There are several advantages to placing the svg element directly in the code over pasting in . 
Although not provided in our project, I use svg elements in such a way that I can influence the parameters of the svg element directly, for example: color, value, background fill and so on.
All assets are stored in the public/pictures folder.

File: frontend/src/components/FollowStar/followStar.jsx [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+ * const [isFavorite, setIsFavorite] = useState(false);
+ * return <FollowStar value={isFavorite} setter={setIsFavorite} />;
+ */
+function FollowStar({ participantId, value, setter }) {
+    const handleToggle = () => {
+        setter(!value);
+    };
+
+    return (
+        <button
```
Comment (heeylena) [inline frontend/src/components/FollowStar/followStar.jsx:25]: For the <button>, you can add an aria-label — an accessibility attribute that explains its purpose for screen readers.

File: frontend/src/pages/Home/home.jsx [hunk 12]
```diff
@@ -1,13 +1,270 @@
...
+                            aria-label="Дізнатися детальніше про компанію Craftmerge"
+                        >
+                            Детальніше про нас
+                        </Button>
+                    </div>
+                </div>
+                <div className={'hero--pictures'}>
+                    <picture>
+                        <source media="(max-width: 769px)" srcSet="/pictures/png/hero-small.png" />
+                        <source media="(max-width: 1513px)" srcSet="/pictures/png/hero-middle.png" />
```
Comment (heeylena) [inline frontend/src/pages/Home/home.jsx:207]: For images, you can also add the WebP format to ensure faster web page loading.

**Review by heeylena — APPROVED** [review]

**Comment by mehalyna** [comment]
The failure in job 47818532960 is due to the use of np.float_, which was removed in NumPy 2.0. The error message from the logs is:

> AttributeError: np.float_ was removed in the NumPy 2.0 release. Use np.float64 instead.

**Solution:**  
Search your codebase for any usage of np.float_ and replace it with np.float64.

**Code Suggestion:**
Replace:
```python
some_variable = np.float_(value)
```
With:
```python
some_variable = np.float64(value)
```

**Action Steps:**
1. Search for np.float_ in your repository (especially in any Python files referenced in your workflow).
2. Replace every instance with np.float64.
3. Re-run your workflow to confirm the solution.

If you need help locating the specific file or line, let me know and I can assist further.
